code/src/main/kotlin/com/expediagroup/sdk/lodgingconnectivity/graphql/BaseGraphQLClient.kt

```diff
@@ -57,16 +61,21 @@ internal class BaseGraphQLClient(config: ExpediaGroupClientConfiguration) : Grap
      * @return The result of the query execution, with errors handled.
      * @throws ExpediaGroupServiceException If the query execution returns errors.
      */
-    override fun <T : Query.Data> execute(query: Query<T>): T {
+    override fun <T : Query.Data> execute(query: Query<T>): RawResponse<T> {
         return runBlocking {
             apolloClient.query(query).execute().apply {
                 if (exception != null) {
                     throw ExpediaGroupServiceException(exception?.message)
```

Comment:
Let's keep the cause itself instead of only copying the message.

The same comment applies to the mutation method.

Reply:
Done